### PR TITLE
Fix some errors and warnings

### PR DESCRIPTION
**Pull request by bakawun**
Compiles succesfully.

### CODE REVIEW

File: gui/source/download.cpp [hunk 1]
```diff
@@ -572,8 +572,6 @@ void checkBootstrapVersion(void){
 	
 	if (res == -1) settings_releasebootstrapver = "No version available";
 	
-	fclose(VerFile);
```
Comment (Jolty95) [inline gui/source/download.cpp]: This must be here. If !Verfile, file doesn't close at all 👎

File: gui/source/download.cpp [hunk 2]
```diff
@@ -622,8 +620,6 @@ void checkBootstrapVersion(void){
 	}
 	if (res == -1) settings_unofficialbootstrapver = "No version available";
 
-	fclose(VerFile);
```
Comment (Jolty95) [inline gui/source/download.cpp]: Same as before 👎

File: gui/source/gamecard.cpp [hunk 3]
```diff
@@ -231,7 +231,6 @@ static void gamecardCacheCTR(void)
 	// TODO: Make sure we don't have more than two lines here.
 	smdh->titles[lang].publisher[0x40-1] = 0;
 	card_text.push_back(utf16_to_wstring(smdh->titles[lang].publisher));
-	free(smdh);
```
Comment (Jolty95) [inline gui/source/gamecard.cpp]: Why do you want to delete this?
calloc MUST be free'd!

File: gui/source/main.cpp [hunk 4]
```diff
@@ -1323,7 +1323,7 @@ int main()
 	//mkdir("sdmc:/_nds/twloader/tmp", 0777);
 	if (logEnabled)	LogFM("Main.Directories", "Directories are made, or already made");
 	
-	snprintf(settings_vertext, 14, "Ver. %d.%d.%d   ", VERSION_MAJOR, VERSION_MINOR, VERSION_MICRO);
+	snprintf(settings_vertext, 13, "Ver. %d.%d.%d   ", VERSION_MAJOR, VERSION_MINOR, VERSION_MICRO);
```
Comment (Jolty95) [inline gui/source/main.cpp]: I don't think this needs any modification....
Have you tested it using a server, uploading a test ver file, etc?
Don't send PR's without testing, especially when it's about update methods

File: gui/source/download.cpp (hunk 1 above)
Comment (bakawun) [inline gui/source/download.cpp]: If the verfile exists it is closed on line https://github.com/Robz8/TWLoader/blob/master/gui/source/download.cpp#L562

File: gui/source/download.cpp (hunk 2 above)
Comment (bakawun) [inline gui/source/download.cpp]: If the verfile exists it is closed on line https://github.com/Robz8/TWLoader/blob/master/gui/source/download.cpp#L601

File: gui/source/download.cpp (hunk 1 above)
Comment (Jolty95) [inline gui/source/download.cpp]: No. It's close if enter in else.

File: gui/source/download.cpp (hunk 2 above)
Comment (Jolty95) [inline gui/source/download.cpp]: Same as before.

File: gui/source/gamecard.cpp (hunk 3 above)
Comment (bakawun) [inline gui/source/gamecard.cpp]: The problem is this free is freeing smdh even when it has already been freed before like on line https://github.com/Robz8/TWLoader/blob/master/gui/source/gamecard.cpp#L215

Comment (Jolty95) [inline gui/source/gamecard.cpp]: I understand you, but it's the same as the fclose before. It only free if https://github.com/Robz8/TWLoader/blob/master/gui/source/gamecard.cpp#L207 is true

File: gui/source/download.cpp (hunk 1 above)
Comment (Jolty95) [inline gui/source/download.cpp]: Ready to merge.

File: gui/source/download.cpp (hunk 2 above)
Comment (Jolty95) [inline gui/source/download.cpp]: Ready to merge.

File: gui/source/gamecard.cpp (hunk 3 above)
Comment (Jolty95) [inline gui/source/gamecard.cpp]: I still thinks this free doesn't need to be there. Maybe with an if clause we can see if was previously free'd or not.

**Pull request merged**